### PR TITLE
docs: translation for reusing logic with custom hooks

### LINKED ISSUE #394: Reusing Logic with Custom Hooks
Page: https://id.react.dev/learn/reusing-logic-with-custom-hooks

Assignee: @rvndaputra

### PR DESCRIPTION
**Pull request by rvndaputra**
Translation to Bahasa for Reusing Logic with Custom Hooks.

Issue: #330

Closes #394

### CODE REVIEW

**Comment by r17x** [comment]
Any updates @rvndaputra ?

**Comment by rvndaputra** [comment]
hi @r17x sorry for the delay. i had to go on a business trip. however, i've already worked on it and it's now ready for review. thank you

File: src/content/learn/reusing-logic-with-custom-hooks.md [hunk 1]
```diff
@@ -1870,27 +1869,27 @@ html, body { min-height: 300px; }
...
-- All Hooks re-run every time your component re-renders.
-- The code of your custom Hooks should be pure, like your component's code.
-- Wrap event handlers received by custom Hooks into Effect Events.
-- Don't create custom Hooks like `useMount`. Keep their purpose specific.
-- It's up to you how and where to choose the boundaries of your code.
+- Custom Hook memungkinkan Anda untuk berbagi logika antar komponen.
+- Custom Hook harus diberi nama yang dimulai dengan `use` diikuti oleh huruf kapital.
+- Custom Hook hanya berbagi logika yang berhubungan dengan state, bukan state itu sendiri.
+- Anda dapat meneruskan nilai reaktif dari satu Hook ke Hook lainnya, dan nilai tersebut tetap up-to-date.
+- Semua Hook dijalankan ulang setiap kali komponen Anda dirender ulang.
```
Comment (r17x) [inline src/content/learn/reusing-logic-with-custom-hooks.md:1880]:
```suggestion
- Semua Hook dijalankan ulang setiap kali komponen Anda di-render ulang.
```

File: src/content/learn/reusing-logic-with-custom-hooks.md [hunk 2]
```diff
@@ -2250,11 +2249,11 @@ export function useInterval(onTick, delay) {
 
 <Solution>
 
-Inside `useInterval`, wrap the tick callback into an Effect Event, as you did [earlier on this page.](/learn/reusing-logic-with-custom-hooks#passing-event-handlers-to-custom-hooks)
+Di dalam `useInterval`, bungkus callback tick menjadi sebuah Event Effect, seperti yang Anda lakukan [sebelumnya di halaman ini.](/learn/reusing-logic-with-custom-hooks#passing-event-handlers-to-custom-hooks)
 
-This will allow you to omit `onTick` from dependencies of your Effect. The Effect won't re-synchronize on every re-render of the component, so the page background color change interval won't get reset every second before it has a chance to fire.
+Ini akan memungkinkan Anda menghilangkan `onTick` dari dependensi Effect Anda. Effect tidak akan disinkronkan setiap kali komponen dirender ulang, sehingga interval perubahan warna latar belakang halaman tidak akan diatur ulang setiap detik sebelum sempat aktif.
```
Comment (r17x) [inline src/content/learn/reusing-logic-with-custom-hooks.md:2254]:
```suggestion
Ini akan memungkinkan Anda menghilangkan `onTick` dari *dependensi* Effect Anda. Effect tidak akan disinkronkan setiap kali komponen di-*render* ulang, sehingga interval perubahan warna latar belakang halaman tidak akan diatur ulang setiap detik sebelum sempat aktif.
```

File: src/content/learn/reusing-logic-with-custom-hooks.md [hunk 3]
```diff
@@ -2321,21 +2320,21 @@ export function useInterval(callback, delay) {
...
 
-In fact, there are five (!) different red dots being rendered. You don't see them because currently they all appear at the same position. This is what you need to fix. What you want to implement instead is a "staggered" movement: each dot should "follow" the previous dot's path. For example, if you quickly move your cursor, the first dot should follow it immediately, the second dot should follow the first dot with a small delay, the third dot should follow the second dot, and so on.
+Faktanya, ada lima (!) titik merah yang berbeda yang dirender. Anda tidak melihat semuanya karena saat ini mereka semua muncul di posisi yang sama. Inilah yang perlu Anda perbaiki. Yang ingin Anda implementasikan adalah gerakan "bergiliran": setiap titik harus "mengikuti" jalur titik sebelumnya. Misalnya, jika Anda memindahkan kursor dengan cepat, titik pertama harus segera mengikuti dengan segera, titik kedua harus mengikuti titik pertama dengan penundaan kecil, titik ketiga harus mengikuti titik kedua, dan seterusnya.
```
Comment (r17x) [inline src/content/learn/reusing-logic-with-custom-hooks.md:2327]:
```suggestion
Faktanya, ada lima (!) titik merah yang berbeda yang di-*render*. Anda tidak melihat semuanya karena saat ini mereka semua muncul di posisi yang sama. Inilah yang perlu Anda perbaiki. Yang ingin Anda implementasikan adalah gerakan "bergiliran": setiap titik harus "mengikuti" jalur titik sebelumnya. Misalnya, jika Anda memindahkan kursor dengan cepat, titik pertama harus segera mengikuti dengan segera, titik kedua harus mengikuti titik pertama dengan penundaan kecil, titik ketiga harus mengikuti titik kedua, dan seterusnya.
```

File: src/content/learn/reusing-logic-with-custom-hooks.md [hunk 4]
```diff
@@ -439,17 +438,17 @@ function Form() {
...
 
-When you need to share the state itself between multiple components, [lift it up and pass it down](/learn/sharing-state-between-components) instead.
+Ketika Anda perlu membagikan state itu sendiri antara beberapa komponen, [angkat dan lewatkan ke bawah](/learn/sharing-state-between-components) sebagai gantinya.
 
-## Passing reactive values between Hooks {/*passing-reactive-values-between-hooks*/}
+## Mengirimkan nilai reaktif antara Hooks {/*passing-reactive-values-between-hooks*/}
 
-The code inside your custom Hooks will re-run during every re-render of your component. This is why, like components, custom Hooks [need to be pure.](/learn/keeping-components-pure) Think of custom Hooks' code as part of your component's body!
+Kode di dalam Custom Hooks akan dijalankan kembali setiap kali komponen Anda dirender ulang. Oleh karena itu, seperti halnya komponen, Custom Hooks [harus bersifat murni.](/learn/keeping-components-pure) Bayangkan kode Custom Hooks sebagai bagian dari badan komponen Anda!
```
Comment (r17x) [inline src/content/learn/reusing-logic-with-custom-hooks.md:449]:
```suggestion
Kode di dalam kustom Hooks akan dijalankan kembali setiap kali komponen Anda di-*render* ulang. Oleh karena itu, seperti halnya komponen, Custom Hooks [harus bersifat murni.](/learn/keeping-components-pure) Bayangkan kode kustom Hooks sebagai bagian dari badan komponen Anda!
```

File: src/content/learn/reusing-logic-with-custom-hooks.md [hunk 5]
```diff
@@ -919,11 +918,11 @@ export function useChatRoom({ serverUrl, roomId, onReceiveMessage }) {
       onMessage(msg);
     });
     return () => connection.disconnect();
-  }, [roomId, serverUrl]); // ✅ All dependencies declared
+  }, [roomId, serverUrl]); // ✅ Semua dependensi dideklarasikan
 }
 ```
 
-Now the chat won't re-connect every time that the `ChatRoom` component re-renders. Here is a fully working demo of passing an event handler to a custom Hook that you can play with:
+Sekarang obrolan tidak akan terhubung kembali setiap kali komponen `ChatRoom` di-render ulang. Berikut adalah demo yang berfungsi penuh untuk meneruskan event handler ke custom Hook yang dapat Anda mainkan:
```
Comment (r17x) [inline src/content/learn/reusing-logic-with-custom-hooks.md:925]: di-render should be italic di-*render*

**Review by r17x — CHANGES_REQUESTED** [review]

**Review by r17x — APPROVED** [review]

**Pull request merged**